### ISSUE TITLE
Import README from repositories

### ISSUE THREAD

**Issue by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#409]
Cache it in our database, and refresh it every time we ingest.

**Comment by orta** [comment]
What I did for CP is to keep a copy on S3, and store the URL to that in the db. Then the website grabs the contents from S3 when showing the page

Looks like this: 
![Screen Shot 2020-06-17 at 4 49 39 PM](https://user-images.githubusercontent.com/49038/84949086-abbe4300-b0ba-11ea-9d2d-5eb8ea33ee01.png)

**Comment by finestructure** [comment]
Just had a quick look into this.

The README url is not part of the repository JSON we're already pulling down.

That brings issues, because GH supports a number of ways to provide a readme. The basename must be `readme` (case insensitive) with an extension for a number of file formats. For Markdown, these could be `.md`, `.markdown` plus a number of others.

What that means is that we can't just pick up the raw content via a regular GET of a url. Instead we need to go via the readme API endpoint:

```
❯ curl https://api.github.com/repos/finestructure/rester/readme
{
  "name": "README.md",
  "path": "README.md",
  "sha": "6a3e0fe6c29f0a07e0e3a27d2466741ffee938c9",
  "size": 4574,
  "url": "https://api.github.com/repos/finestructure/Rester/contents/README.md?ref=develop",
  "html_url": "https://github.com/finestructure/Rester/blob/develop/README.md",
  "git_url": "https://api.github.com/repos/finestructure/Rester/git/blobs/6a3e0fe6c29f0a07e0e3a27d2466741ffee938c9",
  "download_url": "https://raw.githubusercontent.com/finestructure/Rester/develop/README.md",
  "type": "file",
  "content": "IyBS...gYAo=\n",
  "encoding": "base64",
  "_links": {
    "self": "https://api.github.com/repos/finestructure/Rester/contents/README.md?ref=develop",
    "git": "https://api.github.com/repos/finestructure/Rester/git/blobs/6a3e0fe6c29f0a07e0e3a27d2466741ffee938c9",
    "html": "https://github.com/finestructure/Rester/blob/develop/README.md"
  }
}
```

Which is fine, except we'll run into rate limiting issue again unless we reduce our refresh frequency - or move to GraphQL (#284).

**Comment by daveverwer** [comment]
> or move to GraphQL

In case it makes the decision easier, we are also potentially interested in finding the location of the license file too (from #717)

**Comment by finestructure** [comment]
Yeah, I think there's no way around tackling that next :)

**Comment by finestructure** [comment]
I've chucked this into the new GaphQL project but as far as I can tell there's actually no way to get the README via GraphQL. Still, moving the other queries to GraphQL would free up API call budget to make this request, so it's tied into it.

**Comment by finestructure** [comment]
This is now going to be quite straightforward to implement, because it's practically identical to the license fetch, implemented here: #753

**Comment by finestructure** [comment]
Going to close this to record current status and progress: we're now storing the Github url in the db for fetching. In case we find display and loading to have performance implications we'll address that separately.
